lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -66,18 +64,27 @@ import Cardano.Wallet.Primitive.AddressDerivation.Icarus
     ( IcarusKey )
 import Cardano.Wallet.Primitive.AddressDerivation.Shelley
     ( ShelleyKey, toRewardAccountRaw )
-import Cardano.Wallet.Primitive.CoinSelection
-    ( CoinSelection (..), feeBalance )
+import Cardano.Wallet.Primitive.CoinSelection.MA.RoundRobin
+    ( SelectionCriteria (..)
+    , SelectionLimit (..)
+    , SelectionResult (changeGenerated, inputsSelected, outputsCovered)
```

Comment:
Another sign that it might be simpler to make a dedicated `Selection` type, instead of parameterizing `SelectionResult`?

Reply:
No, it is a sign that `DuplicateRecordFields` is a pain in the ass :upside_down_face:

Reply:
> No, it is a sign that `DuplicateRecordFields` is a pain in the ass upside_down_face

Understood.